typescript/core/src/McapStreamReader.ts

```diff
@@ -89,7 +91,41 @@ export default class McapStreamReader {
     if (this.#doneReading) {
       throw new Error("Already done reading");
     }
-    this.#buffer.append(data);
+    this.#appendOrShift(data);
+    this.#reader.reset(this.#view);
+  }
+  #appendOrShift(data: Uint8Array): void {
```

Comment:
Curious what the motivation was for merging this logic into the McapStreamReader – it reduces encapsulation a bit, but maybe there is a perf win? (if so, do we know what the specific source of the win is?)